shared/js/blobview.js

```diff
@@ -376,30 +288,31 @@ var BlobView = (function() {
       if (len % 2) {
         fail('len must be a multiple of two');
       }
+
+      var bytes = new Uint8Array(this.buffer, offset + this.viewOffset, len);
+
       if (le === null || le === undefined) {
-        var BOM = this.getUint16(offset);
-        len -= 2;
-        offset += 2;
+        var BOM = (bytes[0] << 8) + bytes[1];
+        bytes = bytes.subarray(2);
+
         if (BOM === 0xFEFF) {
           le = false;
-        }
-        else {
+        } else if (BOM === 0xFFFE) {
           le = true;
+        } else {
+          fail('expected BOM');
```

Comment:
You've added a new failure case, so this isn't the same as the old implementation. Could be a source of bugs.  Probably better would be to use le=true in this case, but not do the subarray, so the first two bytes don't get truncated.

What would be best is is TextDecoder supports "utf-16" and can handle the BOM automatically. Then all you have to do is pick the encoding based on the le argument.

Or, another option is to just not bother with TextDecoder here, since this is infrequently used and probably pretty fast, and it can't really fail the way it is written.